a3fe/run/calculation.py

```diff
@@ -143,7 +141,7 @@ def _validate_input(self) -> None:
         # Check backwards, as we care about the most advanced preparation stage
         for prep_stage in reversed(_PreparationStage):
             files_absent = False
-            for leg_type in Calculation.required_legs:
+            for leg_type in [_LegType.FREE, _LegType.BOUND]:
```

Comment:
Could we avoid hard-coding this? I'm imagining the case where someone only wants to run e.g. the free leg for some reason. I know that it's hard because we don't yet have the sys prep config, which would contain this information. Some options: 1) supply the sys prep config to __init__ instead of setup 2) Only validate when someone attempts to run setup. What do you think?